x-pack/docs/en/security/es-security-principles.asciidoc

```diff
@@ -0,0 +1,55 @@
+[[es-security-principles]]
+== {es} security principles
+Protecting your {es} cluster and the data it contains is of utmost importance.
+There are numerous security practices and considerations for keeping your
+cluster safe, such as not saving passwords in plain text files or making sure
+that users have appropriate permissions (most users shouldn't be a superuser).
+These guidelines are important, but the following information provides something
+greater. These principles are foundational to running {es} in a secure manner.
+
+[discrete]
+[[security-dont-run-without]]
+=== Run {es} with security enabled
+Never run an {es} cluster without security enabled. This principle cannot be
+overstated. Running {es} without security leaves your cluster exposed to anyone
+on the internet, permitting anyone to download, modify, or delete any data in
+your cluster.
+<<configuring-stack-security,Start the {stack} with security enabled>> or
+<<manually-configure-security,manually configure security>> to prevent
+unauthorized access to your clusters and ensure that internode communication
+is secure. 
+
+[discrete]
+[[security-dont-expose]]
+=== Protect {es} from public internet traffic
+Even with security enabled, never expose {es} to public internet traffic.
+Using an application to sanitize requests to {es} still poses risks, such as
+a malicious user writing <<search,`_search`>> requests that could overwhelm an
+{es} cluster and bring it down. If you need to connect devices from the internet
+to {es}, use some combination of a proxy, firewall, and VPN. The best
+combination of these technologies depends on your security needs and risks.
```

Comment:
Not exposing ES to the internet is another defense in depth thing - sure, security should stop folks from reading it without a password, but the internet is not a safe place. Layers.

Now, talking about a proxy bothers me because it makes me think of something like [squid](https://wiki.squid-cache.org/SquidFaq/ReverseProxy). It's a proxy, but it's not going to give you any security if you put it between ES and the internet. Not if you let `_search` through. Our advice below about using an application to mediate access to ES is really the right way to go here.

Even if your users aren't the whole internet, the application mediating searches is very good. It basically forces you to code review all the requests you'll send to ES. That's good for stability. Our old advice mentioned sanitizing requests. I think this is ultimately pretty futile and I'd drop it. You won't sanitize everything.

Sometimes an admin will have to "go around" your application to do some one off stuff. Or a script will do it for them. And *that* is where the VPN/firewall/jump box/terraform stuff comes into play. I'm not sure it's worth calling out specific solutions because they will feel out of date pretty quick. Separate networks with a firewall was the was we isolated Oracle from our application servers years ago. I imagine the cool kids are doing this with k8s configuration now.

Just talk about keeping ES network isolated from stuff that shouldn't see it, I think. Again, this is just a layer. Layer upon layer of paranoia gives you more confidence.

Reply:
Maybe being more generic in this regard is the best path forward. We could replace the last sentence with something like:
>Keep {es} as isolated as possible, preferably behind a firewall and a VPN.